### PR TITLE
Update gitignore, flake8, and pre-commit config files

### PR DESCRIPTION
**Pull request by willingc**
Addresses #78.

This PR updates and documents the following configuration files:
- .gitignore (use standard Python gitignore and move pyosMeta specific items to end)
- .flake8 (explain ignored rules and add a code complexity check)
- .precommit.config.yaml (improve comments for maintainability and new contributor friendliness)

### CODE REVIEW

File: .gitignore
Comment (lwasser) [inline .gitignore:110]:
```suggestion

```
@willingc i had intentionally removed a lot from the .gitignore because we don't use a lot of this such as pdm. Do you think it's better practice to just include the stock .gitignore as is rather than being intentional about what would actually potentially be used / ignored in our workflow? i'm just curious. this definitely had some things i had to add back such and code coverage files. but then it has other things like this that we just don't need because we don't use it and would never have a pdm toml file in our build or dev workflow using hatch.

File: .gitignore
Comment (lwasser) [inline .gitignore:113]:
```suggestion

```
similarly i think this pep isn't used? or that was my understanding. pdm supports it but i had heard it never was adopted (i could be wrong)

File: .gitignore
Comment (lwasser) [inline .gitignore:120]:
```suggestion

```

File: .gitignore
Comment (lwasser) [inline .gitignore:139]:
```suggestion

```

we don't use mkdocs or rope

File: .gitignore
Comment (lwasser) [inline .gitignore:182]:
```suggestion
src/pyosmeta/_version.py
*.pickle
```

This is my mess 😄 i have learned to use a global .gitignore for dummy test files :)  so we can clean this up too as i have a better workflow now!

File: .pre-commit-config.yaml
Comment (lwasser) [inline .pre-commit-config.yaml:9]: Thank you this is wonderful!

**Review by lwasser — APPROVED** [review]
@willingc thank you for this. see what you think about removing some of the things we don't need or let me know if you think we should keep the default as is. once we clean this up we can merge this. approving!

File: .gitignore
Comment (willingc) [reply .gitignore:110]: I think it is personal preference. I tend to use what is the common standard that GH uses. Your call. I'm cool with whatever. I do like to have pyosMeta specific stuff at the end :smile:

**Pull request merged**
